File: app/src/main/java/com/imangazalievm/bubbble/di/modules/ShotZoomPresenterModule.java
package com.imangazalievm.bubbble.di.modules;

import com.imangazalievm.bubbble.di.scopes.Presenter;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;

@Module
public class ShotZoomPresenterModule {

    private final String shotTitle;
    private final String shotUrl;
    private final String imageUrl;

    public ShotZoomPresenterModule(String shotTitle, String shotUrl, String imageUrl) {
        this.shotTitle = shotTitle;
        this.shotUrl = shotUrl;
        this.imageUrl = imageUrl;
    }

    @Provides
    @Presenter
    @Named("shot_title")
    public String getShotTitle() {
        return shotTitle;
    }

    @Provides
    @Presenter
    @Named("shot_url")
    public String getShotUrl() {
        return shotUrl;
    }

    @Provides
    @Presenter
    @Named("image_url")
    public String getImageUrl() {
        return imageUrl;
    }

}

File: app/src/main/java/com/imangazalievm/bubbble/di/shotzoom/ShotZoomPresenterModule.kt
package com.imangazalievm.bubbble.di.shotzoom

import com.imangazalievm.bubbble.di.global.scopes.Presenter
import dagger.Module
import dagger.Provides
import javax.inject.Named

@Module
class ShotZoomPresenterModule(
    private val shotTitle: String,
    private val shotUrl: String,
    private val imageUrl: String
) {

    @Provides
    @Presenter
    @Named("shot_title")
    fun getShotTitle(): String {
        return shotTitle;
    }

    @Provides
    @Presenter
    @Named("shot_url")
    fun getShotUrl(): String {
        return shotUrl;
    }

    @Provides
    @Presenter
    @Named("image_url")
    fun getImageUrl(): String {
        return imageUrl;
    }

}